src/Core/MailTemplate/Transformation/HTMLToTextTransformation.php

```diff
@@ -47,6 +47,12 @@ public function __construct()
      */
     public function apply($templateContent, array $templateVariables)
     {
+        // The input could be in HTML or raw text. If it is HTML it starts with <!DOCTYPE
+        // As heuristic we interpret the first character and skip transformation if this looks
+        // like text already.
+        if (strlen($templateContent) > 0 && substr($templateContent, 0, 1) != '<') {
```

Comment:
> What if a text starts with < it's technicaly not impossible, not very common I give you that but not impossible.

Now, it is not working at all for text layouts. That's a pragmatic solution to make the Twig mail themes usable if you have text layouts.

If it should be 100% correct, it would be possible to add the layout name or type to the template variables and check for that.

Reply:
Making it 100% perfect seems a bit overengineering here.... Maybe an incremental improvement is acceptable also?